File: 0001/TwoSum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
    int *result = (int*) malloc(sizeof(int)*2);
    int didFound = -1;
    for (int i = 0; i<numsSize; i++) {
        if (didFound == 0) {
            break;
        }

        for (int j = i+1; j<numsSize; j++) {
            if ((nums[i] + nums[j]) == target) {
                result[0] = i;
                result[1] = j;

                didFound = 0;
                break;
            }
        }
    }

    return result;
}

File: 0001/TwoSum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map: [Int: Int] = [:]
        for (index, number) in nums.enumerated() {
            let diff = target - number
            if let hashedIndex = map[diff], hashedIndex != index {
                return [hashedIndex, index]
            }
            map[number] = index
        }
        return []
    }
}